core-inline-blocks/inline-image.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+export const settings = {
+	id: 'inline-image',
+
+	title: __( 'Inline Image' ),
+
+	type: 'image',
+
+	icon: 'format-image',
+
+	render( { id, url, alt, width } ) {
+		const imgWidth = width > 150 ? 150 : width;
+		// set width in style attribute to prevent Block CSS from overriding it
+		const img = `<img class="wp-image-${ id }" style="width:${ imgWidth }px;" src="${ url }" alt="${ alt }" />`;
+
+		return img;
```

Comment:
Would be interesting if this is a React component and then render to string in RichText? Not sure though. Cc @youknowriad @aduth regarding RichText state and formats.

Reply:
Agreed. @iseulde added `edit` and `save` functions that return React elements in https://github.com/WordPress/gutenberg/pull/6959/files#diff-157ef14c5b252d176bfb2fa95850763bR20. Happy to merge those changes if we want, but I think maybe further development is moving to https://github.com/WordPress/gutenberg/pull/6959 now.